torch/_tensor_str.py

```diff
@@ -190,6 +190,8 @@ def _tensor_str(self, indent):
         return '[]'
 
     summarize = self.numel() > PRINT_OPTS.threshold
+    if self.dtype is torch.float16:
+        self = self.float()
```

Comment:
this will give incorrect print.
float16 -> float32 is lossy.

Reply:
I think float16 -> float32 is exact. For real numbers it's just zero-extended exponent and fraction.

The rest of the change lgtm too. I tried out the patch locally and it works well.